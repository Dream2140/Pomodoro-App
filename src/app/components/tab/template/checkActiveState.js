/**
 * @description check active tabs on report page
 * @exports checkActiveState
 * @param {boolean} period check if reports by time should be shown
 * @param {string} value current tab value
 * @return {string} active css class
 */
export default function (period, value) {
    if (period) {
      const regex = /(^\/reports\/)(.*)\//gm;
      const match = regex.exec(`${window.location.pathname}`);
      let time;
  
      if (match) {
        time = match[2];
      }
  
      if (time === value) {
        return 'tab__btn--active';
      }
    } else {
      const regex = /(^\/reports\/)(.*)\/(.*)$/gm;
      const match = regex.exec(`${window.location.pathname}`);
      let category;
  
      if (match) {
        category = match[3];
      }
  
      if (category === value) {
        return 'tab__btn--active';
      }
    }
  }