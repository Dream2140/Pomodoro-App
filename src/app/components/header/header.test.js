/* eslint-disable no-undef */
import checkActiveState from './template/checkActivePage';
import headerTemplate from './template/header.hbs';

it('renders correctly', () => {
    const tree = JSON.stringify(headerTemplate());
    expect(tree).toMatchSnapshot();
});

describe('handlebars helpers', () => {
    let windowSpy;

    beforeEach(() => {
        windowSpy = jest.spyOn(window, 'window', 'get');
        windowSpy.mockImplementation(() => ({
            location: {
                pathname: '/settings/pomodoros',
            },
        }));
    });

    afterEach(() => {
        windowSpy.mockRestore();
    });

    it('should return active btn css class', () => {
        const result = checkActiveState('settings');
        const truthyResult = 'navigation__btn--active';

        expect(truthyResult).toEqual(result);
    });

    it('should be undefined.', () => {
        const result = checkActiveState('wrong data');

        expect(result).toBeUndefined();
    });
});
