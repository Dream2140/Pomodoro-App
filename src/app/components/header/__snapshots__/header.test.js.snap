// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"\\"<header class=\\\\\\"header \\\\\\">\\\\r\\\\n    <div class=\\\\\\"wrapper\\\\\\">\\\\r\\\\n        <div class=\\\\\\"header__container\\\\\\">\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__logo logo\\\\\\">\\\\r\\\\n                <a href=\\\\\\"#\\\\\\" class=\\\\\\"logo__link\\\\\\">\\\\r\\\\n                    <img src=\\\\\\"/images/header.svg\\\\\\" alt=\\\\\\"header\\\\\\" class=\\\\\\"logo__img\\\\\\">\\\\r\\\\n                </a>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <div class=\\\\\\"header__title-wrapp\\\\\\">\\\\r\\\\n                <h1 class=\\\\\\"header__title\\\\\\">Daily task list</h1>\\\\r\\\\n            </div>\\\\r\\\\n\\\\r\\\\n            <nav class=\\\\\\"navigation header__nav\\\\\\">\\\\r\\\\n                <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-tasks\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\" data-route=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon icon-statistics\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                    <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n                        <button class=\\\\\\"navigation__btn \\\\\\"\\\\r\\\\n                            data-route=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n                            <span class=\\\\\\"navigation__icon icon  icon-settings\\\\\\"></span>\\\\r\\\\n                        </button>\\\\r\\\\n                    </li>\\\\r\\\\n                </ul>\\\\r\\\\n            </nav>\\\\r\\\\n\\\\r\\\\n        </div>\\\\r\\\\n\\\\r\\\\n    </div>\\\\r\\\\n</header>\\""`;
